File: src/main/java/no/fintlabs/integration/model/integration/Integration.java
package no.fintlabs.integration.model.integration;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Integration {

    public enum State {
        ACTIVE,
        DEACTIVATED
    }

    private Long sourceApplicationId;
    private String sourceApplicationIntegrationId;
    private String destination;
    private Integration.State state;
    private String activeConfigurationId;

}

File: src/main/kotlin/no/novari/flyt/configuration/model/integration/Integration.kt
package no.novari.flyt.configuration.model.integration

data class Integration(
    var sourceApplicationId: Long? = null,
    var sourceApplicationIntegrationId: String? = null,
    var destination: String? = null,
    var state: State? = null,
    var activeConfigurationId: String? = null,
) {
    enum class State {
        ACTIVE,
        DEACTIVATED,
    }

    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var sourceApplicationId: Long? = null
        private var sourceApplicationIntegrationId: String? = null
        private var destination: String? = null
        private var state: State? = null
        private var activeConfigurationId: String? = null

        fun sourceApplicationId(sourceApplicationId: Long?) = apply { this.sourceApplicationId = sourceApplicationId }

        fun sourceApplicationIntegrationId(sourceApplicationIntegrationId: String?) =
            apply { this.sourceApplicationIntegrationId = sourceApplicationIntegrationId }

        fun destination(destination: String?) = apply { this.destination = destination }

        fun state(state: State?) = apply { this.state = state }

        fun activeConfigurationId(activeConfigurationId: String?) =
            apply { this.activeConfigurationId = activeConfigurationId }

        fun build(): Integration =
            Integration(
                sourceApplicationId = sourceApplicationId,
                sourceApplicationIntegrationId = sourceApplicationIntegrationId,
                destination = destination,
                state = state,
                activeConfigurationId = activeConfigurationId,
            )
    }
}
